BuildingThemes/DetoursHolder.cs

```diff
@@ -1,48 +1,168 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Text;
 using System.Reflection;
+using System.Threading;
 using ColossalFramework;
 using ColossalFramework.Math;
 using UnityEngine;
 
 namespace BuildingThemes
 {
- 
+
     class DetoursHolder
     {
 
+        private static Dictionary<ulong, ushort> seedTable = new Dictionary<ulong, ushort>(); 
+
+        public static void InitTable()
+        {
+
+            for (ushort _seed = 0; _seed <= 65534; ++_seed) { 
+                var seed = (ulong) (6364136223846793005L*(long) _seed + 1442695040888963407L);
+                seedTable.Add(seed, _seed);
+            }
+        }
+
+        private static readonly object Lock = new object();
+
         //we'll use this variable to pass position to GetRandomBuildingInfo method. Or we can just pass District
-        private static Vector3 position;
-        
+        public class Position
```

Comment:
Right now the ``onCalc...LevelUp()`` are setting the ``position`` variable all the time. It never happens that it is ``null``.